hooks/pre-commit

```diff
@@ -1,9 +1,42 @@
-#! /bin/bash
+#!/bin/bash
```

Comment:
Can you please briefly:
 - Describe the problem with the previous "obvious" pre-commit hook.
 - Outline how your new code fixes this problem.

We're doing a lot of extra stuff now, and I'm happy for it provided we justify why we should be doing so. It will also make it possible for us to fix this problem in a different way in the future potentially. At present, you merely state what you're doing, not why.